benchmark/infrastructure/BenchmarkMeasurementContainer.cpp

```diff
@@ -68,38 +68,35 @@ ResultGroup::operator std::string() const {
   std::ostringstream stream;
 
   /*
-  If the given vector is empty, add ` None` to the stream. If it isn't, adds
-  "\n\n" and then calls the given function.
+  If the given vector is empty, return ` None`. If it isn't, return the
+  contacination of "\n\n" and then the string list representation of the vector.
   */
-  auto addWithFunctionOrNone = [&stream](const auto& vec, auto func) {
-    if (!vec.empty()) {
-      stream << "\n\n";
-      func();
-    } else {
-      stream << " None";
-    }
+  auto vectorToStringListOrNone = []<typename T>(const std::vector<T>& vec) {
+    return (vec.empty()
+                ? " None"
+                : absl::StrCat(
+                      "\n\n",
+                      addIndentation(
+                          ad_utility::listToString(
+                              ad_utility::transform(vec,
+                                                    [](const auto& pointer) {
+                                                      return (*pointer);
+                                                    }),
+                              [](const auto& entry) {
+                                return static_cast<std::string>(entry);
+                              },
```

Comment:
Here you don't need to materialize the vecotr, you can just use `std::views::transform`.
And merge the two lambdas into one (this works with the default argument for the `listToString` function that I have suggested.
And maybe first assemble the string via `addIndentation` and THEN call `StrCat` to make this more readable.

Reply:
Partly implemented in [14dd8e1](https://github.com/ad-freiburg/qlever/pull/1010/commits/14dd8e152187c2947da08691269746edbba71179).

I'm not really sure, what you meant with 
> And maybe first assemble the string via addIndentation and THEN call StrCat to make this more readable.

?

Reply:
I meant that you have many functions nested here.
Easier to read would be:
```
auto innerString = listToString(......);

return absl::StrCat("\n\n", innerString, "\n\n");

Reply:
Unfortunately, the string produced from that code is not the same, as the old one. But I tried to do something in the same vein in [8ecd8e9](https://github.com/ad-freiburg/qlever/pull/1010/commits/8ecd8e994771d83e668c5e8ec27336744f3632fe).